pkg/operator/kubeletversionskewcontroller/kubelet_version_skew_controller.go

```diff
@@ -0,0 +1,163 @@
+	return c
+}
+
+func minUpgradeableKubeletSkewForOCPVersion(v string) int {
+	switch semver.MustParse(v).Minor % 2 {
+	case 0: // even OCP versions
+		return 0
+	case 1: // odd OCP versions
+		return -1
+	default:
+		panic("should not happen")
+	}
+}
+
+type kubeletVersionSkewController struct {
+	factory.Controller
+	operatorClient     v1helpers.OperatorClient
+	nodeLister         corev1listers.NodeLister
+	apiServerVersion   semver.Version
+	minUpgradeableSkew int
+}
+
+func (c *kubeletVersionSkewController) sync(_ context.Context, _ factory.SyncContext) error {
+	operatorSpec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	if !management.IsOperatorManaged(operatorSpec.ManagementState) {
+		return nil
+	}
+
+	nodes, err := c.nodeLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	var errorMsgs []string
+	var skewedToTheLimitMsgs []string
+	var skewedButOKMsgs []string
+	var syncedMsgs []string
+	var unsupportedMsgs []string
+
+	// for each node, check kubelet version
+	for _, node := range nodes {
+		kubeletVersion, err := nodeKubeletVersion(node)
+		if err != nil {
+			errorMsgs = append(errorMsgs, fmt.Sprintf("Unable to determine kubelet version on node %s: %s", node.Name, err))
+			continue
+		}
+		skew := int(kubeletVersion.Minor - c.apiServerVersion.Minor)
+		switch {
+		case skew == 0: // synced
+			syncedMsgs = append(syncedMsgs, fmt.Sprintf("Kubelet version on node %s is in synced with the expected API server version.", node.Name))
+		case skew < c.minUpgradeableSkew: // upgrading to next minor version of API server would result in an unsupported config
+			skewedToTheLimitMsgs = append(skewedToTheLimitMsgs, fmt.Sprintf("Kubelet version (%s) on node %s is behind the expected API server version (%s) by at least the maximum allowable skew.", kubeletVersion, node.Name, c.apiServerVersion))
+		case skew < 0: // behind, but upgrading to next minor version of API server is supported
+			skewedButOKMsgs = append(skewedButOKMsgs, fmt.Sprintf("Kubelet version (%s) on node %s is behind the expected API server version (%s), but within the maximum allowable skew.", kubeletVersion, node.Name, c.apiServerVersion))
+		default: // kubelet version newer than api server version. possibly in the middle of a rollback.
+			unsupportedMsgs = append(unsupportedMsgs, fmt.Sprintf("Kubelet version (%s) on node %s is ahead of the expected API server version (%s).", kubeletVersion, node.Name, c.apiServerVersion))
+		}
+	}
+
+	msgs := append(errorMsgs, append(skewedToTheLimitMsgs, append(skewedButOKMsgs, append(syncedMsgs, unsupportedMsgs...)...)...)...)
+	// sort messages to avoid flapping. sort by node name for aesthetics.
+	sort.Sort(byNode(msgs))
+	msg := strings.Join(msgs, "\n")
+
+	condition := operatorv1.OperatorCondition{Type: KubeletVersionSkewLimitUpgradeableConditionType, Message: msg}
```

Comment:
For clusters with hundreds of nodes, this is going to be hard to read if we include all the messages.  Can we drop `msg` and swap in a join of the most-severe message slice in the `switch` below?  E.g.:

```go
case len(errorMsgs) > 0:
	condition.Reason = KubeletVersionUnknownReason
	condition.Status = operatorv1.ConditionFalse
	sort.Sort(byNode(errorMsgs)) // avoid flapping.  Sort by node name for aesthetics
	condition.Message = strings.Join(errorMsgs, "\n")
```